pvar.c

```diff
@@ -4422,14 +4422,14 @@ void pv_value_destroy(pv_value_t *val)
 int pv_printf_s(struct sip_msg* msg, pv_elem_p list, str *s)
 {
 	static int buf_itr = 0;
-	static char buf[PV_PRINT_BUF_NO][PV_PRINT_BUF_SIZE];
+	char buf[PV_PRINT_BUF_NO][pv_print_buf_size];
```

Comment:
@NormB this seems very dangerous as "buf" is assigned to s->s and "returned" by this function.  If "buf" is on the stack instead of the heap, there is a very good chance that this buffer will be corrupted.  This change needs to be reverted unless I'm missing something.